### PR TITLE
[SPARK-8136][YARN] Fix flakiness in YarnClusterSuite.

### PR DESCRIPTION
**Pull request by harishreedharan**
Instead of actually downloading the logs, just verify that the logs link is actually
a URL and is in the expected format.

### CODE REVIEW

**Comment by sryza** [comment]
Does this comment still apply: `// If we are running in yarn-cluster mode, verify that driver logs are downloadable.`?

File: yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala [hunk 1]
```diff
@@ -351,11 +352,13 @@ private object YarnClusterDriver extends Logging with Matchers {
       assert(driverLogs.size === 2)
       assert(driverLogs.containsKey("stderr"))
       assert(driverLogs.containsKey("stdout"))
-      val stderr = driverLogs("stderr") // YARN puts everything in stderr.
-      val lines = Source.fromURL(stderr).getLines()
```
Comment (srowen) [inline yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala]: LGTM; `Source` isn't used now is it?

Comment (harishreedharan) [inline yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala]: Yep, I will clean the import up. Also, I think I am going to remove the check for hostname (just rely on the URL format).

**Comment by harishreedharan** [comment]
`YarnClusterSuite` has passed, running it again to ensure it is not flaky.

Jenkins, retest this please.

**Comment by harishreedharan** [comment]
Jenkins, test this please.

**Comment by harishreedharan** [comment]
Jenkins, test this please.

**Comment by andrewor14** [comment]
looks great, thanks for fixing this. I'll merge this once tests pass.

**Comment by harishreedharan** [comment]
Looks like `YarnClusterSuite` has passed.

**Comment by harishreedharan** [comment]
This failed in streaming tests. I think this is good for commit, since the YARN tests all passed

**Comment by andrewor14** [comment]
I just merged this in master. However it doesn't merge cleanly in 1.4 and 1.3. Could you open PRs against those branches?

**Comment by harishreedharan** [comment]
I will open a PR against 1.4 (if this functionality was committed to that branch). Since this functionality and the test is not in 1.3 branch, it should not be needed in that branch.

**Comment by harishreedharan** [comment]
So this feature was not in 1.3 or 1.4 branches. So we are good.

**Comment by andrewor14** [comment]
Alright sounds good

**Comment by andrewor14** [comment]
Could you close this PR then? It seems that github is not closing them after merge for some reason.

**Comment by harishreedharan** [comment]
Closing since this was merged into master.

**Pull request closed without merging**